cmd/note_common.go

```diff
@@ -166,42 +165,64 @@ func createCommitComments(project string, mrID int, commit string, body string,
 			}
 		} else if strings.HasPrefix(scanner.Text(), "|") {
 			if comments != "" {
-				createCommitNote(project, mrID, commit, newfile, oldfile, oldLineNum, newLineNum, comments, block)
+				createCommitNote(project, mrID, commit, newfile, oldfile, linetype, oldLineNum, newLineNum, comments, block)
 				comments = ""
 			}
 			// read line numbers
-			fs := strings.Split(scanner.Text(), " ")
+			f := strings.Split(strings.Join(strings.Fields(strings.TrimSpace(scanner.Text())), " "), " ")
+			if f[1] == "@@" {
+				continue;
+			}
 
 			oldLineNum = -1
 			newLineNum = -1
-			for _, f := range fs {
-				if f == "" || f == "|" || f == "@@" {
-					continue
-				}
-				val, err := strconv.Atoi(f)
-				if err != nil {
-					// NaN
-					if strings.HasPrefix(f, "+") {
-						newLineNum = oldLineNum
-						oldLineNum = -1
+
+			// The output can be, for example:
+			// | # # [no +/-] < context comment
+			// | # #    +     < newline comment
+			// | # #    -     < oldline comment
+			// | #      -     < oldline comment
+			// |   #    +     < newline comment
+
+			// f[0] is always a "|"
+			// f[1] will always be a number
+			val1, _ := strconv.Atoi(f[1])
+			val2, err := strconv.Atoi(f[2])
+			if err == nil { // f[2] is a number
+				if len(f) > 3 { // f[3] exists
+					newLineNum = val2
+					// f[3] starts with a +
+					if strings.HasPrefix(f[3], "+") {
```

Comment:
the last two comments are too obvious IMO :(

Reply:
Yeah, it's more for me to read through TBH ;).  I'd prefer to keep them in place for now.